locale.c

```diff
@@ -6105,68 +6105,66 @@ S_my_langinfo_i(pTHX_
 #    if      defined(HAS_SNPRINTF)                                          \
        && (! defined(HAS_LOCALECONV) || defined(TS_W32_BROKEN_LOCALECONV))
 
-        {
```

Comment:
I expect removing the braces here would break C++ builds, assuming both this case and the following case can be compiled.

Removing the braces changes the scope of the declarations in the block from being limited to the block to include past the next case, which allows the initializations (like for floatbuf) to be bypassed, which isn't valid in C++.

Reply:
I wasn't able to find a case where both blocks are compiled on Linux, but they don't seem to be entirely excluded.

Reply:
Sorry, I had already figured this out, but forgot I already had an open pull request.  And, it turns out that there was another place that needed the toggling.  PR now updated